.travis.yml

```diff
@@ -82,27 +76,38 @@ before_install:
   - echo $LC_ALL
   - $CXX -v
   - export PYTHONPATH=$PYTHONPATH:$INSTALL_DIR/lib/python/;
-  - if [ "$NRN_ENABLE_PYTHON_DYNAMIC" == "ON" ]; then
-      pyenv global 2.7.15 3.7.1;
-    else
-      pyenv global $PYTHON_VERSION;
-    fi
 install:
+  - $PIP install --user setuptools;
   - $PIP install --user scipy matplotlib bokeh ipython cython --upgrade;
+  # add cython location to PATH
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+      export CYTHON_LIB=$($PIP show cython | grep Location);
+      export CYTHON_EXE=$(echo $CYTHON_LIB | awk '{print $2}' | awk -F "lib" '{print$1}')"bin";
+      export PATH=$CYTHON_EXE:$PATH;
+      echo $PATH;
+    fi
 
 #=============================================================================
 # Build, install and test
 #=============================================================================
 script:
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+      export CONFIG_OPTIONS="$CONFIG_OPTIONS --with-readline=no";
+    fi
   - if [ "$BUILD_MODE" == "cmake" ]; then
       if [ "$NRN_ENABLE_PYTHON_DYNAMIC" == "ON" ]; then
-        export PYTHON2=$(pyenv which python2.7);
-        export PYTHON3=$(pyenv which python3.7);
-        CMAKE_OPTION="-DNRN_ENABLE_PYTHON=ON -DNRN_ENABLE_PYTHON_DYNAMIC=ON -DNRN_PYTHON_DYNAMIC=${PYTHON2};${PYTHON3}";
+        export PYTHON2=$(which python2);
+        export PYTHON3=$(which python3);
+        export CMAKE_OPTION="-DNRN_ENABLE_PYTHON=ON -DNRN_ENABLE_PYTHON_DYNAMIC=ON -DNRN_PYTHON_DYNAMIC=${PYTHON2};${PYTHON3}";
+      else
+        export CMAKE_OPTION="$CMAKE_OPTION -DPYTHON_EXECUTABLE=${PYTHON}";
       fi;
       mkdir build && cd build;
       cmake $CMAKE_OPTION -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR ..;
       cmake --build . -- -j 2;
+      if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+        echo $'[install]\nprefix='>src/nrnpython/setup.cfg;
```

Comment:
CMAKE: Workaround for an error during python installation in mac
error: must supply either home or prefix/exec-prefix -- not both

Reply:
@WeinaJi : we are not using homebrew installed on travis OSX, right? because this workaround typically required if python is installed via homebrew.

Reply:
Now as I get rid of pyenv, I use the default python version in the traivs system. I guess the python is pre-installed via homebrew in travis osx because previously with pyenv I had to  "pyenv install $PYTHON_VERSION" myself.

Reply:
Fro the path, it seems system python and not homebrew? You can try removing the logic to create setup.cfg